Plugins/Data/NWScript/inc_array.nss

```diff
@@ -0,0 +1,485 @@
+        ExecuteStatement("INSERT INTO "+GetTableName(tag, obj)+" VALUES ( " + IntToString(index) + ", '" + element + "')", obj);
+    }
+}
+
+void Array_Insert_Flt(string tag, int index, float element, object obj=OBJECT_INVALID)
+{
+    Array_Insert_Str(tag, index, FloatToString(element), obj);
+}
+
+void Array_Insert_Int(string tag, int index, int element, object obj=OBJECT_INVALID)
+{
+    Array_Insert_Str(tag, index, IntToString(element), obj);
+}
+
+void Array_Insert_Obj(string tag, int index, object element, object obj=OBJECT_INVALID)
+{
+    Array_Insert_Str(tag, index, ObjectToString(element), obj);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Insert a new element at the end of the array.
+void Array_PushBack_Str(string tag, string element, object obj=OBJECT_INVALID)
+{
+    if (obj == OBJECT_INVALID)
+        obj = GetModule();
+    int rowCount = GetRowCount(tag, obj);
+    ExecuteStatement("INSERT INTO "+GetTableName(tag, obj)+" VALUES ( " + IntToString(rowCount) + ", '" + element + "')", obj);
+}
+
+void Array_PushBack_Flt(string tag, float element, object obj=OBJECT_INVALID)
+{
+    Array_PushBack_Str(tag, FloatToString(element), obj);
+}
+
+void Array_PushBack_Int(string tag, int element, object obj=OBJECT_INVALID)
+{
+    Array_PushBack_Str(tag, IntToString(element), obj);
+}
+
+void Array_PushBack_Obj(string tag, object element, object obj=OBJECT_INVALID)
+{
+    Array_PushBack_Str(tag, ObjectToString(element), obj);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Cuts the array off at size 'size'.  Elements beyond size are removed.
+void Array_Resize(string tag, int size, object obj=OBJECT_INVALID)
+{
+    if (obj == OBJECT_INVALID)
+        obj = GetModule();
+    ExecuteStatement("delete from "+GetTableName(tag, obj)+" where ind >= " + IntToString(size), obj);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+void Array_Shuffle(string tag, object obj=OBJECT_INVALID)
+{
+    if (obj == OBJECT_INVALID)
+        obj = GetModule();
+
+    string table = GetTableName(tag, obj);
+    ExecuteStatement("CREATE TABLE " +table+ "_temp AS SELECT ROW_NUMBER() OVER(ORDER BY RANDOM())-1, value FROM " +table, obj);
+    ExecuteStatement("DELETE FROM " +table , obj);
+    ExecuteStatement("INSERT INTO " +table+ " SELECT * FROM " +table+ "_temp", obj);
+    ExecuteStatement("DROP TABLE " +table+ "_TEMP", obj);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+int Array_Size(string tag, object obj=OBJECT_INVALID)
+{
+    return GetRowCount(tag, obj);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Sort the array by value according to 'direciton' (ASC or DESC)
+// Note that this is a lexical sort, so sorting an array of ints or floats will have
+// odd results
```

Comment:
Is this true? If so, I don't think it's sufficient, and we should create the temp table with the correct type so it's not 'odd'.